### DISCUSSION TITLE
Proposed Feature: Mass Move Checked Media to a Different Path in Scenes View

### DISCUSSION THREAD

**Discussion by neox1969** [stashapp/stash#5583] [Ideas]
Feature Overview: The ability to mass-move selected media files to a different storage path directly within the Scenes View offers a more streamlined and efficient workflow for organizing and managing media. This feature would allow users to select multiple files, define a new or existing storage path, and automatically update the database with the new file locations. This functionality would be invaluable for organizing large libraries of media, ensuring files are properly categorized in specific folders, and maintaining a consistent organizational structure.

Key Benefits:

1.	Time Savings: Mass movement of media eliminates the need for manual file transfers, which can be tedious and time-consuming when managing large libraries.
2.	Improved Organization: Users can quickly group and categorize media into specific paths that align with their workflow or storage schema.
3.	Database Synchronization: The feature ensures that the database reflects updated paths automatically, reducing the risk of broken file links or misplaced media.
4.	Enhanced Usability: Offering flexible options for selecting, tagging, and moving files caters to diverse organizational needs.
________________________________________
Proposed Workflow and Options:

When working in the Scenes View, users would access the mass-move functionality through a menu option under the ellipsis (three dots) or a context menu. The following options and submenus would be available:
1. Move Selected Files
•	Move to Path:
   o	Users can select an existing folder or path where the checked files will be moved.
   o	The system will provide a list of current paths available in the storage directory.
   o	Once selected, the files will be moved to the designated folder, and the database will be updated with the new file locations.
•	Create New Path and Move Files:
   o	This option allows users to define a new folder or path directly within the interface.
   o	Once the new path is created, the selected files will be moved there, and the database will be updated accordingly.
________________________________________
2. Move Specific Tagged Media
This advanced option filters the files to be moved based on metadata tags, enabling precise targeting for organizing files. The sub-options could include:

•	By Director:
   o	Select files tagged with specific directors’ names and move them to a chosen or newly created path.
•	By Galleries:
   o	Organize files from specific galleries into defined folders.
•	By Studio:
   o	Move media associated with particular studios to a designated path.
•	By Performers:
   o	Filter and move files tagged with one or more performers.
•	By Groups:
   o	Select and organize files based on custom group tags.
•	By Tags:
   o	Utilize custom or system-defined tags to filter files for movement.

For each of the above filters:
•	Choose Path:
   o	Users can select from existing paths where the filtered files will be moved.
•	Create New Path and Move Files:
   o	Users can create a new path and move the filtered files directly into this folder.
________________________________________
User Interface Design Suggestions:
1.	Checkbox Selection:
   o	Implement checkboxes next to each media file to allow multi-selection directly in the Scenes View.
2.	Path Selection Dialog:
   o	Provide a clear dialog box for choosing or creating new paths, with options for viewing and organizing current folder hierarchies.
3.	Progress Indicator:
   o	Show a progress bar or indicator while the files are being moved to provide feedback and ensure a seamless user experience.
4.	Confirmation and Undo Option:
   o	After moving files, prompt the user with a confirmation dialog and offer an “Undo” option to reverse the action if needed.
________________________________________
Conclusion: The Mass Move Checked Media feature would enhance organizational efficiency by simplifying the process of transferring multiple files to different paths while ensuring the database stays up-to-date. By incorporating flexible tagging and path creation options, users can manage their media libraries effectively with minimal effort. This feature is especially beneficial for users dealing with extensive media collections, providing a powerful tool for structured file management.

**Comment by feederbox826** [comment]
already handled by one of many renamer/move plugins

https://github.com/stashapp/CommunityScripts/tree/main/plugins/renamerOnUpdate
